### PR TITLE
Null course ID #4184

### LINKED ISSUE #4184: Null course ID
java.lang.AssertionError: null course id
	at teammates.common.util.Assumption.fail(Assumption.java:59)
	at teammates.common.util.Assumption.assertTrue(Assumption.java:27)
	at teammates.common.util.Assumption.assertNotNull(Assumption.java:269)
	at teammates.ui.controller.InstructorFeedbackEditCopyAction.execute(InstructorFeedbackEditCopyAction.java:28)
	at teammates.ui.controller.Action.executeAndPostProcess(Action.java:337)
	at teammates.ui.controller.ControllerServlet.doPost(ControllerServlet.java:59)
	at teammates.ui.controller.ControllerServlet.doGet(ControllerServlet.java:38)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:617)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:717)
	at org.mortbay.jetty.servlet.ServletHolder.handle(ServletHolder.java:511)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1166)
	at teammates.storage.datastore.DatastoreFilter.doFilter(DatastoreFilter.java:28)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.appengine.tools.appstats.AppstatsFilter.doFilter(AppstatsFilter.java:142)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.ParseBlobUploadFilter.doFilter(ParseBlobUploadFilter.java:125)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.runtime.jetty.SaveSessionFilter.doFilter(SaveSessionFilter.java:35)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.JdbcMySqlConnectionCleanupFilter.doFilter(JdbcMySqlConnectionCleanupFilter.java:60)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.TransactionCleanupFilter.doFilter(TransactionCleanupFilter.java:43)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at org.mortbay.jetty.servlet.ServletHandler.handle(ServletHandler.java:388)
	at org.mortbay.jetty.security.SecurityHandler.handle(SecurityHandler.java:216)
	at org.mortbay.jetty.servlet.SessionHandler.handle(SessionHandler.java:182)
	at org.mortbay.jetty.handler.ContextHandler.handle(ContextHandler.java:765)
	at org.mortbay.jetty.webapp.WebAppContext.handle(WebAppContext.java:418)
	at com.google.apphosting.runtime.jetty.AppVersionHandlerMap.handle(AppVersionHandlerMap.java:257)
	at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:152)
	at org.mortbay.jetty.Server.handle(Server.java:326)
	at org.mortbay.jetty.HttpConnection.handleRequest(HttpConnection.java:542)
	at org.mortbay.jetty.HttpConnection$RequestHandler.headerComplete(HttpConnection.java:923)
	at com.google.apphosting.runtime.jetty.RpcRequestParser.parseAvailable(RpcRequestParser.java:76)
	at org.mortbay.jetty.HttpConnection.handle(HttpConnection.java:404)
	at com.google.apphosting.runtime.jetty.JettyServletEngineAdapter.serviceRequest(JettyServletEngineAdapter.java:146)
	at com.google.apphosting.runtime.JavaRuntime$RequestRunnable.run(JavaRuntime.java:442)
	at com.google.tracing.TraceContext$TraceContextRunnable.runInContext(TraceContext.java:437)
	at com.google.tracing.TraceContext$TraceContextRunnable$1.run(TraceContext.java:444)
	at com.google.tracing.CurrentContext.runInContext(CurrentContext.java:230)
	at com.google.tracing.TraceContext$AbstractTraceContextCallback.runInInheritedContextNoUnref(TraceContext.java:308)
	at com.google.tracing.TraceContext$AbstractTraceContextCallback.runInInheritedContext(TraceContext.java:300)
	at com.google.tracing.TraceContext$TraceContextRunnable.run(TraceContext.java:441)
	at com.google.apphosting.runtime.ThreadGroupPool$PoolEntry.run(ThreadGroupPool.java:234)
	at java.lang.Thread.run(Thread.java:745)

### PR DESCRIPTION
**Pull request by LowWeiLin**
Fixes #4184

### CODE REVIEW

**Comment by thyageshm** [comment]
hey there are other error pages o be handled. like: entitydoesnotexist. or do you think that should be a separate issue?

**Comment by LowWeiLin** [comment]
The other pages are not defined as error pages in web.xml which was why I did not add status codes to them. I suppose they should be handled though.

**Comment by LowWeiLin** [comment]
Updated

File: src/main/webapp/entityNotFoundPage.jsp [hunk 1]
```diff
@@ -1,4 +1,5 @@
 <%@ page import="teammates.common.util.Const"%>
+<% response.setStatus(500);%>
```
Comment (thyageshm) [inline src/main/webapp/entityNotFoundPage.jsp]: maybe consider 404?

File: src/main/webapp/unauthorized.jsp [hunk 2]
```diff
@@ -1,4 +1,5 @@
 <%@ page import="teammates.common.util.Const"%>
+<% response.setStatus(403);%>
```
Comment (thyageshm) [inline src/main/webapp/unauthorized.jsp]: I think this one should be 401

**Comment by thyageshm** [comment]
Sorry for being picky... I looked at the list here: https://en.wikipedia.org/wiki/List_of_HTTP_status_codes

Made some minor comments in the code based on the numbers to use.

After that, just make sure the tests pass and you can merge :-)

File: src/main/webapp/unauthorized.jsp (hunk 2 above)
Comment (LowWeiLin) [inline src/main/webapp/unauthorized.jsp]: web.xml sets it to 403. Should I change that too?

Comment (thyageshm) [inline src/main/webapp/unauthorized.jsp]: yup. the online guide says 401 is for "unauthorised" access

File: src/main/webapp/entityNotFoundPage.jsp (hunk 1 above)
Comment (LowWeiLin) [inline src/main/webapp/entityNotFoundPage.jsp]: 404 is for page not found. Does not seem right to use it?

**Comment by LowWeiLin** [comment]
Had to leave 403 in web.xml, it is thrown on unauthorized access to admin pages.

**Comment by thyageshm** [comment]
ok. if nothing else, you can merge

**Pull request merged**